mesoscope/cellpack
# add `batch_job_id` to result metadata

Problem
=======
What is the problem this work solves, including
closes #296 

Solution
========
What I/we did to solve this problem
- batch job ids are stored as environment tokens, we can retrieve them via `os`
- add a `batch_job_id` field to result metadata, which will be uploaded to firebase upon successful packing. If the packing is local, the value will be `null`

with @ascibisz 

## Type of change
Please delete options that are not relevant.
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. use API gateway to trigger a packing batch job (currently, only Alli and Ruge have access, but I'll attach screenshots to demonstrate the steps )
2. wait 30-60sec for the job to run
3. check the `results` collection in staging firebase database - recipe metadata should display the correct batch job id  
4. optional: pack a recipe locally - the `batch_job_id` field should also be present but with a `null` value


Screenshots (optional):
-----------------------
Show-n-tell images/animations here
Step 1 (API Gateway): use API gateway to trigger a packing batch job
<img width="500" alt="Screenshot 2024-10-28 at 3 27 22 PM" src="https://github.com/user-attachments/assets/d78f68b9-5f09-4177-83eb-96c55a8b5f11">

Step 2 (AWS Batch): wait for the batch job to run
<img width="500" alt="Screenshot 2024-10-28 at 3 29 44 PM" src="https://github.com/user-attachments/assets/64d2d395-c5a7-4515-8396-9e76a9ba539d">

Step 3 (Firebase): batch job id will be added to the metadata and uploaded to firebase
<img width="500" alt="Screenshot 2024-10-28 at 3 30 38 PM" src="https://github.com/user-attachments/assets/0e7afa0c-3db1-4291-9049-abb6fa0fda1e">
